Upstream patch from 50.0 release for CVE-2020-37011

CVE references:
https://nvd.nist.gov/vuln/detail/CVE-2020-37011
https://www.exploit-db.com/exploits/48803

Upstream fix:
https://gitlab.gnome.org/GNOME/gnome-font-viewer/-/commit/841617bd24e4e6329de5c8268aef9f14f1c5ffda

A full update to GNOME Fonts 50.0 is not done here because 50.0
requires libadwaita-1 >= 1.8, while the current Solaris Userland gate
has libadwaita 1.3.6. Keep this as a targeted security fix for the 41.0
component.

From 841617bd24e4e6329de5c8268aef9f14f1c5ffda Mon Sep 17 00:00:00 2001
From: Khalid Abu Shawarib <kas@gnome.org>
Date: Fri, 13 Mar 2026 23:48:45 +0300
Subject: [PATCH] sushi-font-widget: Fix a heap overflow

It was counting characters after terminating the string with a null
character.
---
 src/sushi-font-widget.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/sushi-font-widget.c b/src/sushi-font-widget.c
index c635702..cbedd54 100644
--- a/src/sushi-font-widget.c
+++ b/src/sushi-font-widget.c
@@ -262,9 +262,11 @@ build_charlist_for_face (FT_Face face,
   c = FT_Get_First_Char (face, &glyph);
 
   while (glyph != 0) {
-    g_string_append_unichar (string, (gunichar) c);
+    if (c != 0) {
+      g_string_append_unichar (string, (gunichar) c);
+      total_chars++;
+    }
     c = FT_Get_Next_Char (face, c, &glyph);
-    total_chars++;
   }
 
   if (length)
-- 
GitLab

